lib/spack/spack/cmd/ci.py

```diff
@@ -526,10 +547,8 @@ def ci_rebuild(args):
     if compiler_action != "FIND_ANY":
         install_args.append("--no-add")
```

Comment:
seems like the comment above is wrong?

Reply:
I don't recall why we wouldn't want `--no-add` in all cases.  Since we never encounter this in practice any more, we just end up always adding this flag.  We use the flag because when the purpose of the current job is to install a dependency of one of the roots of the environment, we just want to install it, not add it as a root itself.